### DISCUSSION TITLE
Enabling TSCH in Whitefield

### DISCUSSION THREAD

**Discussion by Jawaheralharbi** [whitefield-framework/whitefield#127] [Q&A]
I am trying to change the mac layer to TSCH. However, the attached error prevented me from that. Also, I don't have superuser permission on the PC that I am using. How can I solve this problem rather than gaining superuser permission?

![snapshot](https://user-images.githubusercontent.com/50404124/129812982-8990cfa3-7813-416b-aa9e-55f5715558a7.png)

**Comment by nyrahul** [comment]
I do not think you will be able to use TSCH without significant changes in Whitefield. TSCH requires channel hopping which is not supported in Whitefield at the moment (though it can be implemented).

Anyways, regarding the problem mentioned above about opening a TUN interface ... you either need SU or CAP_NET_ADMIN cap enabled for the user.
Try to run the contiki-ng bin specified in the config directly from the command line ... if it works there .. it should also work inside of Whitefield.
